src/randomx.cpp

```diff
@@ -400,4 +400,15 @@ extern "C" {
 		machine->run(machine->tempHash);
 		machine->getFinalResult(output, RANDOMX_HASH_SIZE);
 	}
+
+	void randomx_calculate_hash_v2(const void* input, size_t inputSize, const void* v1_in, void* v2_out) {
+		assert(inputSize == 0 || input != nullptr);
+		assert(v1_in != nullptr);
+		assert(v2_out != nullptr);
```

Comment:
Hi, this can help prevent user error when calling the library.
assert(v1_in != v2_out);

Reply:
It will work perfectly fine with `v1_in == v2_out`.